### ISSUE TITLE
libpostal_setup() only works once

### ISSUE THREAD

**Issue by boshkins** [openvenues/libpostal#6]
Al,

libpostal_setup() only works once, all subsequent calls fail, regardless of calls to libpostal_teardown() in between, the error message is:

```
ERR   Error loading transliteration module
   at libpostal_setup (libpostal.c:742) errno: No such file or directory
```

I guess a given app is not likely to call setup/teardown more than once per run, so this is not a big deal. My context is a test suite that tries to separate test cases cleanly, thus multiple setup/teardown call pairs.

Thank you,
Anatoly

**Comment by thatdatabaseguy** [comment]
Fixed with https://github.com/openvenues/libpostal/commit/12816d0e9542423871435397a145a686fc4ffed1

In general libpostal_setup/libpostal_teardown are relatively expensive calls as they load models from disk and are only supposed to run once per application (e.g. in a dynamic language like Python these would run at module import time).

Unless you're testing whether libpostal's setup works (though they're not currently on Github, libpostal has its own tests which can do that), I wouldn't recommend more than one call to setup and one call to teardown in the entire program.

Also note: libpostal is _not_ thread-safe, so if the tests are threaded it might be problematic.

**Comment by boshkins** [comment]
Understood, thank you.
My testing is purely exploratory; double calls to setup() were an artifact
of the test framework I use (Catch). Would be very interested to see your
test suite!


On Mon, Sep 28, 2015 at 5:44 PM, Al Barrentine <notifications@github.com>
wrote:

> Closed #6 <https://github.com/openvenues/libpostal/issues/6>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openvenues/libpostal/issues/6#event-421261600>.
>



-- 
Anatoly "Tolya" Boshkin
Boshkins Software Corporation
email: anatoly@boshkins.com
Skype:anatoly.boshkin
mob phone: (703) 472 0184
